zephyr/samples/modules/mcuboot/mcuboot_external_config/README.rst

```diff
@@ -0,0 +1,30 @@
+.. _mcuboot_external_config:
+
+MCUBoot External Config
+#######################
+
+Overview
+********
+
+A simple sample illustrating how to configure from a different module
+an MCUBoot image, using local keys and certificates.
+
+NOTE: This references ``${ZEPHYR_MCUBOOT_MODULE_DIR}/boot/zephyr``
+for most configuration files for the verified set of boards.  Other
+board configuration files will need to be provided locally.
+
+Building and Running
+********************
+
+This application can be built and executed on QEMU as follows:
+
+.. code-block:: console
+
+  west build -b nrf52840dk_nrf52840 \
+    bootloader/mcuboot/samples/modules/mcuboot/mcuboot_external_config
+
+
+Sample Output
+=============
+
+None.
```

Comment:
Waste of section, if there is nothing to present here.

Reply:
Q: This really has a different purpose than a sample (conceptual, non-error handling) and a test suite (verifying functionality and error handling).  It is a real reference application (configuration, etc).  The closes guidance we have from Zephyr Project is to locate this in `$mcuboot/zephyr/app/` (from Zephyr's example-application), but that directory (tree) naming doesn't scale to support multiple applications.  Where should this reference application configuration go?

Reply:
> This is a standard section in Zephyr documentation. Explicitly stating there is no output is useful information.

Fair point. 

Someday I am going to write sample returning literally `None.`, to make this section confusing.

Reply:
> Q: This really has a different purpose than a sample (conceptual, non-error handling) and a test suite (verifying functionality and error handling). It is a real reference application (configuration, etc). The closest guidance we have from Zephyr Project is to locate this in `$mcuboot/zephyr/app/` (from Zephyr's example-application), but that directory (tree) naming doesn't scale to support multiple applications. Where should this reference application configuration go?

Good question. Maybe we should have some place for "reference board configurations", or templates. We probably lack some configuration management here.

Reply:
The problem with these reference configurations is that they do not explain much, regarding selected Kconfig options or DTS definitions.

Reply:
According to Zephyr's glossary, reference **configurations** are a type of "application", whether or not it also contains local logic or data. The configurations could simply be a configuration of overlays located in other modules.  

The explanations of the Kconfig settings and the overlays needs to be outside the applications.  There needs to be a module-level location for reusable overlays.  We can already use `\${ZEPHYR_<moduleNameUpper>_MODULE_DIR}` in our CONF_FILE assignments _before_ calling `find_project()`.

I would like to see Zephyr Project identify a tree parallel to `samples/` and `tests/` for multiple non-sample, non-test applications.  This could be named `apps/` or `applications/`.  It should not be named with a singular noun (e.g `app/`).